### PR TITLE
Fix comma scaling and add tests

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1923

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs [hunk 1]
```diff
@@ -165,16 +182,21 @@ public static bool IsValidFormatArg(string formatString, CultureInfo formatCultu
                     // Reset hasNumericCharacters to false to later check if any numeric character after decimal point.
                     decimalPointIndex = i;
                     hasNumericCharacters = false;
+
                     if (commaIdxList.Count > 0)
                     {
-                        // Add escaping character for any group separator character ',' before decimal point.
-                        for (int j = commaIdxList.Count - 1; j >= 0; j--)
+                        // Remove any comma before decimal point that is not using for scaling
+                        // if (i > 0 && formatStr[i - 1] != ',')
```
Comment (LucGenetier) [inline src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs:189]: >// if (i > 0 && formatStr[i - 1] != ',')  [](http://example.com/codeflow?start=24&length=41)

remove if not needed

**Review by LucGenetier — DISMISSED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs (hunk 1 above)
Comment (nguhoa) [reply src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs]: remove comment here. thanks

**Review by LucGenetier — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**